src/main/java/uk/gov/hmcts/darts/common/service/impl/RetrieveCoreObjectServiceImpl.java

```diff
@@ -69,17 +73,22 @@ public HearingEntity retrieveOrCreateHearing(String courthouseName, String court
     }
 
     private HearingEntity createHearing(String courthouseName, String courtroomName, String caseNumber, LocalDate hearingDate, UserAccountEntity userAccount) {
-        CourtCaseEntity courtCase = retrieveOrCreateCase(courthouseName, caseNumber, userAccount);
-        CourtroomEntity courtroom = retrieveOrCreateCourtroom(courtCase.getCourthouse(), courtroomName);
+        final CourtCaseEntity courtCase = retrieveOrCreateCase(courthouseName, caseNumber, userAccount);
+        final CourtroomEntity courtroom = retrieveOrCreateCourtroom(courtCase.getCourthouse(), courtroomName);
         HearingEntity hearing = new HearingEntity();
+
         hearing.setCourtCase(courtCase);
         hearing.setCourtroom(courtroom);
         hearing.setHearingDate(hearingDate);
         hearing.setNew(true);
         hearing.setHearingIsActual(false);
         hearing.setCreatedBy(userAccount);
         hearing.setLastModifiedBy(userAccount);
+        hearing.setCreatedDateTime(OffsetDateTime.now());
```

Comment:
The disadvantage of calling the static `OffsetDateTime.now()` method is that we have no way to mock the behaviour of this, which makes it hard to test as this value will be different every time the test runs.

So, if we do need to set the timestamp manually via the setter, it's better to use `currentTimeHelper.currentOffsetDateTime()` to obtain the timestamp, such that `currentTimeHelper` can be mocked in tests with a fixed value.

This comment applies to all uses of `OffsetDateTime.now()` in this PR.